ambry-api/src/main/java/com.github.ambry/rest/AuthorizationService.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.rest;
+
+import com.github.ambry.account.Container;
+import java.util.Collection;
+
+
+/**
+ * A service that handles authorization of requests. This will be provided to the {@link SecurityService} for making
+ * access decisions that might require the help of an external service. The call to {@link #hasAccess(RestRequest)}
+ * should not require expensive network calls, since it will potentially be used on the critical path. Instead,
+ * the {@link AuthorizationService} should maintain a cache of ACLs for the resources it manages permissions on.
+ */
+public interface AuthorizationService {
+
+  /**
+   * Makes an access decision based on the information provided in a {@link RestRequest}.
+   * @param request the request that contains the target resource and information to establish requester identity
+   * @throws RestServiceException if access to a resource is denied.
+   */
+  void hasAccess(RestRequest request) throws RestServiceException;
```

Comment:
That was the idea. Upon discussions with @vgkholla, we decided it might be better for this to return a status code instead of throwing here (which could allow for different reasons for allowing/denying access).